src/language/statics/Info.re

```diff
@@ -739,45 +739,60 @@ let fixed_typ_ok: ok_pat => Typ.t =
   | Ana(Consistent({join, _})) => join
   | Ana(InternallyInconsistent({ana, _})) => ana;
 
-let fixed_typ_err_common: error_common => Typ.t =
-  fun
-  | NoType(FreeConstructor(c)) =>
-    Sum([
-      ConstructorMap.Variant(c, [Id.invalid], None),
-      ConstructorMap.BadEntry(Unknown(Internal) |> Typ.temp),
-    ])
-    |> Typ.temp
-  | NoType(BadToken(_) | BadLabel(_) | InvalidLabel(_)) =>
-    Unknown(Internal) |> Typ.temp
-  | TupleLabelError({typ, _})
-  | DuplicateLabel(_, typ) => typ
-  | Inconsistent(Expectation({ana, _})) => ana
-  | Inconsistent(Internal(_)) => Unknown(Internal) |> Typ.temp // Should this be some sort of meet?
-  | Inconsistent(CompareFun(_)) => Atom(Bool) |> Typ.temp
-  | Inconsistent(WithArrow(_)) =>
-    Arrow(Unknown(Internal) |> Typ.temp, Unknown(Internal) |> Typ.temp)
-    |> Typ.temp;
-
-let fixed_typ_err: error_exp => Typ.t =
-  fun
-  | UnboundLivelit(_)
-  | FreeVariable(_)
-  | UnusedDeferral
-  | BadPartialAp(_)
-  | InexhaustiveMatch(_)
-  | WantTuple
-  | BadOperator(_)
-  | LabelNotFound(_, _)
-  | BadTrivAp(_) => Unknown(Internal) |> Typ.temp
-  | Common(err) => fixed_typ_err_common(err)
-  | InvalidUseMode({inner_typ, _}) => inner_typ
-  | BadLivelitModel(ana) => ana;
-
-let fixed_typ_err_pat: error_pat => Typ.t =
-  fun
-  | ExpectedConstructor
-  | Redundant(_) => Unknown(Internal) |> Typ.temp
-  | Common(err) => fixed_typ_err_common(err);
+let fixed_typ_err_common: (error_common, Typ.t) => Typ.t =
+  (err, ana) =>
+    switch (err) {
+    | NoType(FreeConstructor(c)) =>
+      if (Typ.is_syn_plus(ana)) {
+        Sum([
+          ConstructorMap.Variant(c, [Id.invalid], None),
+          ConstructorMap.BadEntry(Unknown(Internal) |> Typ.temp),
+        ])
+        |> Typ.temp;
+      } else {
+        ana;
```

Comment:
I guess this could also be the join of the ana/sum if they're consistent :shrug: